module-services/service-cellular/src/volte/ImsiParserInterface.hpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include "OperatorInfo.hpp"
+#include <string>
+#include <optional>
+
+namespace cellular::service
+{
+    class ImsiParserInteface
+    {
+      public:
+        virtual ~ImsiParserInteface() = default;
+
+        virtual auto parse(std::string imsi) -> std::optional<OperatorInfo> = 0;
```

Comment:
This is probably okay, thanks to SSO(short/small string optimization) :) If anticipated strings are longer, I'd use move/const reference here.

Reply:
fixed